src/CommonUtilities/DrsUriParser.cs

```diff
@@ -0,0 +1,235 @@
+
+            // Validate accessions/ids
+            if (parsingError is null && !IsCompactIdUriValid(uri))
+            {
+                // This is a hostname URI. However we turned off the host uri parsing in base.InitializeAndValidate (to prevent file-based URIs), so we'll do it here.
+                if (!IsHostnameUriValid(uri))
+                {
+                    parsingError = new UriFormatException("Malformed DRS URI: slashes are not allowed in the DRS ID.");
+                }
+                else
+                {
+                    switch (Uri.CheckHostName(uri.Host))
+                    {
+                        case UriHostNameType.Dns:
+                        case UriHostNameType.IPv4:
+                        case UriHostNameType.IPv6:
+                            break;
+
+                        case UriHostNameType.Basic:
+                        default:
+                            parsingError = new("Invalid URI: The Authority/Host could not be parsed.");
+                            break;
+                    }
+                }
+            }
+
+            static bool IsHostnameUriValid(Uri uri)
+            {
+                if (!uri.IsAbsoluteUri || uri.IsUnc)
+                {
+                    return false;
+                }
+
+                if (new Uri($"{Uri.UriSchemeHttps}{uri.OriginalString[UriSchemeDrs.Length..]}").HostNameType == UriHostNameType.Basic)
+                {
+                    return false;
+                }
+
+                return IsDrsIdValid(uri);
+            }
+
+            static bool IsCompactIdUriValid(Uri uri)
+            {
+                if (!IsCompactIdUri(uri))
+                {
+                    return false;
+                }
+
+                return IsDrsIdValid(uri);
+            }
+
+            static bool IsDrsIdValid(Uri uri)
+            {
+                var path = uri.AbsolutePath;
+
+                // Return false if it appears a compact id URI with provider_code was parsed. Note that hostname Uris don't allow unescaped embedded slashes in the path.
+                return !path.TrimStart('/').Any(c => c == '/') && !path.TrimStart(':').Any(c => c == ':') && IsAccessionValid(uri.Segments.Last().TrimStart(':'));
+            }
+        }
+
+        private static bool IsCompactIdUri(Uri uri)
+            => uri.OriginalString.StartsWith(_prefix, StringComparison.OrdinalIgnoreCase) && _drsCompactId.IsMatch(uri.OriginalString.AsSpan(_prefixLength));
+
+        // https://ga4gh.github.io/data-repository-service-schemas/docs/#tag/DRS-API-Principles/DRS-IDs
+        // https://ga4gh.github.io/data-repository-service-schemas/docs/more-background-on-compact-identifiers.html#tag/Background-on-Compact-Identifier-Based-URIs
+        [GeneratedRegex("\\A(?<provider_code>[\\._a-z]+?/)?(?<namespace>[\\._a-z]+?):(?<accession>[%-\\.0-9A-Z_a-z~]+?)\\Z", RegexOptions.ExplicitCapture | RegexOptions.CultureInvariant)]
```

Comment:
this looks pretty gnarly and potentially buggy... do you think we have enough test cases for it?

Reply:
I have test cases to cover with and without `provider_code`, both passing and failing the allowed character set, and other missing elements. This regex only covers Compact ID DRS URI (both styles).

The regex does not cover Hostname DRS URI, which are covered via a combination of [this code utilizing an `https` Uri](https://github.com/microsoft/ga4gh-tes/blob/79581005b8a52e4eda378a4723b1afcabee50180/src/CommonUtilities/DrsUriParser.cs#L123) and [this code limiting the `Path` portion of the Uri](https://github.com/microsoft/ga4gh-tes/blob/79581005b8a52e4eda378a4723b1afcabee50180/src/CommonUtilities/DrsUriParser.cs#L218) but I also have test cases covering both passing and failing (in similar ways) for those as well.